test/CppTests/Tests/NatvisTests.cs

```diff
@@ -480,6 +480,56 @@ public void TestCommaFormatWithSquareBrackets(ITestSettings settings)
             }
         }
 
+        [Theory]
+        [DependsOnTest(nameof(CompileNatvisDebuggee))]
+        [RequiresTestSettings]
+        public void TestMultipleNatvisFiles(ITestSettings settings)
+        {
+            this.TestPurpose("This test checks if TreeItems are visualized.");
+            this.WriteSettings(settings);
+
+            IDebuggee debuggee = Debuggee.Open(this, settings.CompilerSettings, NatvisName, DebuggeeMonikers.Natvis.Default);
+
+            using (IDebuggerRunner runner = CreateDebugAdapterRunner(settings))
+            {
+                this.Comment("Configure launch");
+                List<string> visFile = new List<string>();
+                visFile.Add(Path.Join(debuggee.SourceRoot, "visualizer_files", "Simple1.natvis"));
+                visFile.Add(Path.Join(debuggee.SourceRoot, "visualizer_files", "Simple2.natvis"));
+
+                LaunchCommand launch = new LaunchCommand(settings.DebuggerSettings, debuggee.OutputPath, visFile, false);
+                runner.RunCommand(launch);
+
+                this.Comment("Set Breakpoint");
+                SourceBreakpoints writerBreakpoints = debuggee.Breakpoints(NatvisSourceName, ReturnSourceLine);
+                runner.SetBreakpoints(writerBreakpoints);
+
+                runner.Expects.StoppedEvent(StoppedReason.Breakpoint).AfterConfigurationDone();
+
+                using (IThreadInspector threadInspector = runner.GetThreadInspector())
+                {
+                    IFrameInspector currentFrame = threadInspector.Stack.First();
+
+                    this.Comment("Verifying TreeItems natvis");
+                    var map = currentFrame.GetVariable("map");
```

Comment:
This only tests to see if `Simple2.natvis` got parsed correctly. Can you also test for `Simple1,natvis`?

What happens if we have collisions in the natvis files.

Reply:
You are only testing map. you are not testing any variable that is being visualized by Simple1.natvis.